Fleece/API_Impl/Fleece.cc

```diff
@@ -719,7 +719,7 @@ FLError FLEncoder_GetError(FLEncoder e) FLAPI {
 }
 
 const char* FL_NULLABLE FLEncoder_GetErrorMessage(FLEncoder e) FLAPI {
-    return e->hasError() ? e->errorMessage.c_str() : nullptr;
+    return e->errorMessage.empty() ? nullptr : e->errorMessage.c_str();
```

Comment:
It appears to me that we are taking advantage of the unsynchronized pair (error_code, error_message) after reset() (we only reset with failure). What's purpose to clear error code but not clear the error message? Or put it another way, why do we have to clear the error code, or can we keep both error code and error message after reset()? 
Is the encoder still usable after reset()?

Reply:
The code implies the encoder is still usable after reset. I don't know if anything is reusing encoders -- I could check our code, but devs using CBL-C also use FLEncoder. So I didn't want to break that functionality.

It's necessary to clear the error code or reusing the encoder will fail immediately. Leaving the message alone seemed safe since no one's going to look at the error message until after there's an error.